### PR TITLE
feat(devices): add onekey support

### PR DESCRIPTION
**Pull request by somebodyLi**

### CODE REVIEW

**Comment by achow101** [comment]
Is there a simulator/emulator for the onekey? Can you add tests so that CI can check against it? See https://github.com/bitcoin-core/HWI/blob/master/test/test_trezor.py and https://github.com/bitcoin-core/HWI/blob/master/test/test_keepkey.py for examples of how to do that.

**Comment by 424778940z** [comment]
Sure, we working on it.

File: hwilib/devices/onekey.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    Any,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+)
+
+py_enumerate = enumerate  # Need to use the enumerate built-in but there's another function already named that
+
+VENDORS = ("onekey.so", "trezor.io",)
```
Comment (prusnak) [inline hwilib/devices/onekey.py:41]: Fix:

```suggestion
VENDORS = ("onekey.so", )
```

File: hwilib/devices/onekey.py [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+        self.onekey_version = onekey_version
+        self.onekey_serial = onekey_serial
+        self.bootloader_version = bootloader_version
+        self.serial_no = serial_no
+        self.boardloader_version = boardloader_version
+
+
+ONEKEY_MAPPING = DEFAULT_MAPPING.register(OnekeyFeatures)
+
+USB_IDS = {(0x1209, 0x4F4A), (0x1209, 0x4F4B), (0x1209, 0x53C1)}
```
Comment (prusnak) [inline hwilib/devices/onekey.py:229]: Fix:

```suggestion
USB_IDS = {(0x1209, 0x4F4A), (0x1209, 0x4F4B), }
```

File: hwilib/udev/51-onekey.rules [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# If you are creating a distribution package,
+# put this into /usr/lib/udev/rules.d or /lib/udev/rules.d
+# depending on your distribution
+
+# OneKey
+# onekey boot
+SUBSYSTEM=="usb", ATTR{idVendor}=="1209", ATTR{idProduct}=="4F4A", MODE="0660", GROUP="plugdev", TAG+="uaccess", TAG+="udev-acl", SYMLINK+="onekey%n"
+# onekey firmware
+SUBSYSTEM=="usb", ATTR{idVendor}=="1209", ATTR{idProduct}=="4F4B", MODE="0660", GROUP="plugdev", TAG+="uaccess", TAG+="udev-acl", SYMLINK+="onekey%n"
+SUBSYSTEM=="usb", ATTR{idVendor}=="1209", ATTR{idProduct}=="53c1", MODE="0660", GROUP="plugdev", TAG+="uaccess", TAG+="udev-acl", SYMLINK+="onekey%n"
```
Comment (prusnak) [inline hwilib/udev/51-onekey.rules:15]: This is a Trezor USB ID, don't include it here:
```suggestion
```

**Review by prusnak — CHANGES_REQUESTED** [review]
I requested some changes below, so your newly added code does not get triggered when a Trezor is connected.

Please connect a Trezor and test whether HWI still works correctly (and Trezor code path is used for communicating with Trezor, not Onekey code path).

**Pull request closed without merging**